web/src/main.rs

```diff
@@ -0,0 +1,81 @@
+mod cron;
+mod err;
+mod jobs;
+mod pages;
+mod routes;
+mod sessions;
+mod templates;
+mod users;
+
+use cja::{app_state::AppState as AS, server::run_server};
+use db::setup_db_pool;
+use miette::IntoDiagnostic;
+
+use tracing::info;
+
+type WebResult<T, E = err::Error> = Result<T, E>;
+
+#[derive(Clone, Debug)]
+struct AppState {
+    db: sqlx::PgPool,
+    cookie_key: cja::server::cookies::CookieKey,
+}
+
+impl AS for AppState {
+    fn db(&self) -> &sqlx::PgPool {
+        &self.db
+    }
+
+    fn version(&self) -> &str {
+        "dev"
+    }
+
+    fn cookie_key(&self) -> &cja::server::cookies::CookieKey {
+        &self.cookie_key
+    }
+}
+
+fn main() -> miette::Result<()> {
+    let _sentry_guard = cja::setup::setup_sentry();
+
+    tokio::runtime::Builder::new_multi_thread()
+        .worker_threads(4)
+        .enable_all()
+        .build()
+        .into_diagnostic()?
+        .block_on(async { _main().await })
+}
+
+async fn _main() -> miette::Result<()> {
```

Comment:
`_main` is just cause our error reporter service `Sentry` wants to be configured BEFORE we start up Tokio, just FYI